src/web/rustdoc.rs

```diff
@@ -259,110 +295,117 @@ impl RustdocPage {
         // while logging OOM errors from html rewriting
         let html = match utils::rewrite_lol(rustdoc_html, max_parse_memory, ctx, templates) {
             Err(RewritingError::MemoryLimitExceeded(..)) => {
                 metrics.html_rewrite_ooms.inc();
 
-                let err = anyhow!(
-                    "Failed to serve the rustdoc file '{}' because rewriting it surpassed the memory limit of {} bytes",
-                    file_path, config.max_parse_memory,
-                );
-
-                ctry!(req, Err(err))
+                return Err(AxumNope::InternalError(
+                    anyhow!(
+                        "Failed to serve the rustdoc file '{}' because rewriting it surpassed the memory limit of {} bytes",
+                        file_path, config.max_parse_memory,
+                    )
+                ));
             }
-            result => ctry!(req, result),
+            result => result.context("error rewriting HTML")?,
         };
 
-        let mut response = Response::with((Status::Ok, html));
-        response.headers.set(ContentType::html());
-        let link_value = LinkValue::new(canonical_url)
-            .push_rel(RelationType::ExtRelType("canonical".to_string()));
-
-        response.headers.set(Link::new(vec![link_value]));
-
-        response.extensions.insert::<CachePolicy>(if is_latest_url {
-            CachePolicy::ForeverInCdn
-        } else {
-            CachePolicy::ForeverInCdnAndStaleInBrowser
-        });
-        Ok(response)
+        Ok((
+            StatusCode::OK,
+            TypedHeader(CanonicalUrl(
+                Uri::builder()
+                    .scheme("https")
+                    .authority("docs.rs")
+                    .path_and_query(canonical_path)
+                    .build()
+                    .context("can't build canonical URL")?,
+            )),
+            Extension(if is_latest_url {
+                CachePolicy::ForeverInCdn
+            } else {
+                CachePolicy::ForeverInCdnAndStaleInBrowser
+            }),
+            Html(html),
+        )
+            .into_response())
     }
 }
 
+#[derive(Clone, Deserialize)]
+pub(crate) struct RustdocHtmlParams {
+    name: String,
+    version: String,
+    // both target and path are only used for matching the route.
+    // The actual path is read from the request `Uri` because
+    // we have some static filenames directly in the routes.
+    #[allow(dead_code)]
+    target: Option<String>,
+    #[allow(dead_code)]
+    path: Option<String>,
+}
+
 /// Serves documentation generated by rustdoc.
 ///
 /// This includes all HTML files for an individual crate, as well as the `search-index.js`, which is
 /// also crate-specific.
-pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
-    let metrics = extension!(req, Metrics).clone();
+#[allow(clippy::too_many_arguments)]
+#[instrument(skip_all)]
```

Comment:
Why is this useful if we're not logging any of the args?

Reply:
Some of these `instrument` calls are leftovers from debugging issues I had. 
In general these are still useful since any logmessage emitted inside this handler or whatever the handler calls will get this context attached.